### ISSUE TITLE
FP for unit-no-unknown when SCSS map key starts with digit

### ISSUE THREAD

**Issue by vilchik-elena** [stylelint/stylelint#4375]
Rule `unit-no-unknown` produces FPs for SCSS map key names starting with number

```scss
$foo: ( 1bar  : "value")
```

Config:
```json
{
  "rules": {
     "unit-no-unknown": true,
  }
}
```

Stylelint version `11.1.1`, checked with https://stylelint.io/demo

I expect no warnings to be flagged.
The following warnings were flagged:

```shell
1:9 error unexpected unknown unit "bar" (unit-no-unknown)
```

**Comment by hudochenkov** [comment]
Thank you for the report!

We should ignore such constructions in `unit-no-unknown`.

**Comment by fanich37** [comment]
If no one is working on it I could check it.

**Comment by hudochenkov** [comment]
@fanich37 go ahead!

**Comment by fanich37** [comment]
@hudochenkov I'm a bit stuck with this issue. It appeared more tricky than I thought.
There is no proper way to understand that we're checking the map. it's possible but looks like a hack for me. The tricky thing is that map could contain another map.
And even if we're sure it's a map we'll need to exclude only its property names from checking. It's possible, but again it seems like a hack.
I'm afraid the rule could become quite difficult only because of map analysis that is not even a part of standard CSS syntax.
I've checked several rules where map issue existed. All the solutions implemented were just ignoring maps from analysis completely. But in tests for this rule we have:
```js
code: '$breakpoints: ( small: 767px, medium: 992pix, large: 1200px );',
```
So I need an advice here. Try to implement map property names analysis or just ignore non-standard syntax properties and update tests (remove the line above)?

**Comment by hudochenkov** [comment]
> ignore non-standard syntax properties

This is the direction we are taking most of the time. And I would go with this one.
